### PR TITLE
Enable learning rate change

### PR DESCRIPTION
**Pull request by xchen010**
I modified the following three files to enable learning rate change during training, similar to darknet. 
eg. 
--lr 1e-4
--steps [100,25000]
--scales [10,0.1]
This gives initial learning rate of 1e-4, then changes it to (1e-4)x10=1e-3 at training step 100, next, update it to (1e-3)x0.1=1e-4 at training step 25000.

### CODE REVIEW

**Comment by abagshaw** [comment]
@xchen010 This has been sitting open for a while and needs to have a conflict resolved before merging. If this is still something that you'd like to have merged could you go through and resolve whatever conflicts are showing up? Thanks!

**Comment by marcunzueta** [comment]
I applied the changes from the commits directly, and I have done a few tests, and it works great! Thank you @xchen010 !
